pkg/webhooks/machine_webhook.go

```diff
@@ -2429,5 +2429,10 @@ func validateAWSInstanceMarketType(providerSpec *machinev1beta1.AWSMachineProvid
 	if providerSpec.MarketType == machinev1beta1.MarketTypeCapacityBlock && len(providerSpec.CapacityReservationID) == 0 {
 		return errors.New("capacityReservationID is required when CapacityBlock is provided")
 	}
+	switch providerSpec.MarketType {
+	case "", machinev1beta1.MarketTypeOnDemand, machinev1beta1.MarketTypeSpot, machinev1beta1.MarketTypeCapacityBlock:
+	default:
+		return fmt.Errorf("invalid marketType: %s Valid values are: %s, %s, %s and omitted", providerSpec.MarketType, machinev1beta1.MarketTypeOnDemand, machinev1beta1.MarketTypeSpot, machinev1beta1.MarketTypeCapacityBlock)
```

Comment:
This would be more consistent with the API errors
```suggestion
		return fmt.Errorf("invalid marketType: %s: valid values are: %s, %s, %s and omitted", providerSpec.MarketType, machinev1beta1.MarketTypeOnDemand, machinev1beta1.MarketTypeSpot, machinev1beta1.MarketTypeCapacityBlock)
```